src/main/kotlin/org/opensearch/replication/task/autofollow/AutoFollowTask.kt

```diff
@@ -65,9 +67,21 @@ class AutoFollowTask(id: Long, type: String, action: String, description: String
     override suspend fun execute(scope: CoroutineScope, initialState: PersistentTaskState?) {
         stat = AutoFollowStat(params.patternName, replicationMetadata.leaderContext.resource)
         while (scope.isActive) {
-            addRetryScheduler()
-            autoFollow()
-            delay(replicationSettings.autofollowFetchPollDuration.millis)
+            try {
+                addRetryScheduler()
+                autoFollow()
+                delay(replicationSettings.autofollowFetchPollDuration.millis)
+            }
+            catch(e: OpenSearchException) {
+                // Any transient error encountered during auto follow execution should be re-tried
+                val status = e.status().status
+                if(status < 500 && status != RestStatus.TOO_MANY_REQUESTS.status) {
+                    log.error("Exiting autofollow task", e)
+                    throw e
+                }
+                log.debug("Encountered transient error while running autofollow task", e)
+                delay(replicationSettings.autofollowFetchPollDuration.millis)
```

Comment:
Shouldn't we have exponential backoff here?

Reply:
Kept it simple for now as poll duration by default is `1min` (see previous comment for index replication task)

Reply:
Sure